interfaces/nginx_route/v0/README.md

```diff
@@ -0,0 +1,60 @@
+# `nginx-route`
+
+## Overview
+
+This relation interface describes the expected behaviour of any charm 
+interfacing with Kubernetes Nginx ingress resources.
+
+## Usage
+
+An application charm can use the `nginx-route` relation to request the creation
+of a Nginx ingress resource in the Kubernetes cluster. The specific 
+configuration of the ingress resources can be controlled by the data provided
+by the relation's requirer. The [`nginx_route` charm library](https://charmhub.io/nginx-ingress-integrator/libraries/nginx_route)
+can be used to handle the provider and requirer sides of this relation.
+
+## Direction
+
+The `nginx-route` interface is a one-way relation. The requirer of this relation
+provides the configuration for the required ingress resources as relation data,
+and the provider does not return anything in the relation.
+
+## Behavior
+
+The requirer and the provider must adhere to a certain set of criteria to be 
+considered compatible with the interface.
+
+### Provider
+
+- Is expected to create ingress resources in the Kubernetes cluster if the requirer's requirements are valid.
+- Is expected to create ingress resources with host-based routing.
+
+### Requirer
+
+- Is expected to provide a valid ingress resource configuration.
+
+## Relation Data
+
+### Provider
+
+The provider does not provide anything in the relation databag.
+
+[\[JSON Schema\]](./schemas/provider.json)
```

Comment:
What does this line do please?

Reply:
I was just following the template, I think this should link to the provider schema definition file.

Reply:
Correct. However, we're migrating to pydantic models. Would you mind replacing the json schema with a pydantic equivalent?

Reply:
Converted to pydantic, thanks!